sanic_routing/route.py

```diff
@@ -221,6 +239,28 @@ def defined_params(self):
     def raw_path(self):
         return self._raw_path
 
+    @property
+    def segments(self) -> t.Tuple[str, ...]:
+        """
+        Same as self.parts except generalized so that any dynamic parts do not
+        include param keys since they have no impact on routing.
+        """
+        return tuple(
+            [
+                f"<__dynamic__:{self._params[idx].label}>"
+                if self._params.get(idx)
+                else segment
+                for idx, segment in enumerate(self.parts)
+            ]
+        )
```

Comment:
I think you can drop the square brackets, just passing a generator expression within tuple(). And perhaps use `in` rather than `get` in the condition?

Reply:
In next push.